examples/openssl_engine_example.cpp

```diff
@@ -0,0 +1,338 @@
+
+        default:
+            goto usage;
+        }
+    }
+
+    if (brokers.empty() || enginePath.empty() || topic_str.empty() || optind != argc) {
+    usage:
+        std::string features;
+        conf->get("builtin.features", features);
+        fprintf(stderr,
+            "Usage: %s [options] -b <brokers> -s <cert-subject> -p <priv-key-password>\n"
+            "\n"
+            "Openssl engine integration example. This example fetches metadata\n"
+            "over ssl channel with broker, established using openssl engine.\n"
+            "\n"
+            "librdkafka version %s (0x%08x, builtin.features \"%s\")\n"
+            "\n"
+            " Options:\n"
+            "  -b <brokers>          Broker address\n"
+            "  -p <engine path>      Path to openssl engine\n"
+            "  -c <ca cert location> File path to ca cert\n"
+            "  -i <engine id>        Openssl engine id\n"
+            "  -e <engine parameter> Openssl engine parameter\n"
+            "  -t <topic_name>       Topic to fetch / produce\n"
+            "  -d [facs..]           Enable debugging contexts: %s\n"
+            "  -X <prop=name>        Set arbitrary librdkafka configuration property\n"
+            "\n",
+            argv[0],
+            RdKafka::version_str().c_str(), RdKafka::version(),
+            features.c_str(),
+            RdKafka::get_debug_contexts().c_str());
+        exit(1);
+    }
+
+    if (conf->set("bootstrap.servers", brokers, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (conf->set("security.protocol", "ssl", errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (conf->set("ssl.engine.location", enginePath, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (ca_location.length() > 0 && 
+        conf->set("ssl.ca.location", ca_location, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (engine_id.length() > 0 &&
+        conf->set("ssl.engine.id", engine_id, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (engine_callback_data.length() > 0 &&
+        conf->set_engine_callback_data((void *) engine_callback_data.c_str(), errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    /* We use the Certificiate verification callback to print the
+     * certificate name being used. */
+    PrintingSSLVerifyCb ssl_verify_cb;
+
+    if (conf->set("ssl_cert_verify_cb", &ssl_verify_cb, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    /*
+     * Create producer using accumulated global configuration.
+     */
+    RdKafka::Producer* producer = RdKafka::Producer::create(conf, errstr);
```

Comment:
`*` goes on the variable, not the type. Throghout.

Reply:
resolved in last iteration